llvm/include/llvm/CodeGen/TargetInstrInfo.h

```diff
@@ -404,6 +408,10 @@ class TargetInstrInfo : public MCInstrInfo {
     return true;
   }
 
+  /// Return true if the instruction should be considered for hoisting by
+  /// MachineLICM, even if it is "cheap".
+  virtual bool canHoistCheapInst(const MachineInstr &MI) const { return false; }
```

Comment:
We already have `TargetInstrInfo::shouldHoist`. Why do we need this one?

Reply:
I think cheap instructions are never hoisted, regardless of what `shouldHoist` says